src/main/java/com/smartbear/plugins/swaggerhub/PublishToHubAction.java

```diff
@@ -105,8 +105,12 @@ private boolean publishApi(RestService restService) throws IOException {
             } else {
                 exporter = new OpenAPI3Exporter(restService.getProject());
             }
-            String result = exporter.exportToFileSystem(Files.createTempDir().getAbsolutePath() + File.separator + "api-docs.json", versionId,
+            String tempDirectoryPath = Files.createTempDir().getAbsolutePath();
+            String tempFilePath = tempDirectoryPath + File.separator + "api-docs.json";
+            String result = exporter.exportToFileSystem(tempFilePath, versionId,
                     "json", new RestService[]{restService}, restService.getBasePath());
+            new File(tempDirectoryPath).deleteOnExit();
+            new File(tempFilePath).deleteOnExit();
```

Comment:
I believe this doesn't work in ReadyAPI for some reason and we have a special tool for that. Use ReadyTools.deleteOnExit().